lisp/ess-r-mode.el

```diff
@@ -2260,7 +2260,40 @@ state.")
         ess-help-sec-regex ess-help-r-sec-regex
         ess-help-sec-keys-alist ess-help-r-sec-keys-alist ; TODO: Still necessary?
         inferior-ess-help-command inferior-ess-r-help-command
-        inferior-ess-help-filetype nil))
+        inferior-ess-help-filetype nil)
+  (when ess-help-add-help-links
+    (ess-r-help-add-links)))
+
+(define-button-type 'ess-r-help-link
+  'follow-link t
+  'action (lambda (_) (ess-r-help-button-action)))
+
+(defun ess-r-help-button-action ()
+  "Display help for button at point."
+  (let ((text (get-text-property (point) 'ess-r-help-link-text)))
+    (ess-display-help-on-object text)))
+
+(defun ess-r-help-add-links ()
+  "Add links to the help buffer."
+  (save-excursion
+    ;; Search for fancy quotes only. If users have
+    ;; options(useFancyQuotes) set to something other than TRUE this
+    ;; probably won't work. If it's FALSE, R outputs ascii ', but
+    ;; searching through the whole buffer takes too long.
+    (while (re-search-forward
+            (rx (minimal-match (and "‘" (1+ not-newline) "’"))) nil t)
+      (let ((inhibit-read-only t)
+            (beg (match-beginning 0))
+            (end (match-end 0))
+            ;; Remove surrounding quotes so the text property is the
+            ;; name of the link:
+            (text (string-trim (match-string 0) (rx (0+ (or "‘" "\""))) (rx (0+ (or "’" "\""))))))
+        ;; Only add links for help topics we can find
+        (when (member text (ess-s-get-help-topics-function ess-local-process-name))
```

Comment:
Do we really need this check? I think occasional false positives are harmless but you can introduce false negative with this. 

The result of  `ess-s-get-helpt-topics-function` should be let-cached locally. It writes to dribble buffer each time it's called. 

I think the defcustom is not warranted either way.

Reply:
We get quite a few false positives without it since we catch anything that's in between fancy quotes. I guess we can try it out that way though during testing and see if it's too annoying.

I removed the defcustom.